core/src/main/java/org/mule/runtime/core/routing/IdempotentMessageFilter.java

```diff
@@ -134,6 +128,27 @@ protected boolean accept(Event event, Event.Builder builder) {
     }
   }
 
+  @Override
+  public final Event process(Event event) throws MuleException {
+    if (accept(event)) {
+      return processNext(event);
+    } else {
+      event.getContext().success();
+      return null;
```

Comment:
One of the things we are considering is to not return null anymore and fail always that the filter does not accept the event. That way we manage that case through the error handler and we avoid weird scenarios with request-response sources and also with some places where the filter is placed and only filters a chain but not the entire flow.